pkg/server/plugin/nodeattestor/aws/iid_test.go

```diff
@@ -0,0 +1,424 @@
+			},
+			expectErr: "client error",
+		},
+		{
+			desc: "non-zero device index",
+			mockExpect: func(mock *mock_aws.MockEC2Client) {
+				output := getDefaultDescribeInstancesOutput()
+				output.Reservations[0].Instances[0].RootDeviceType = &instanceStoreType
+				output.Reservations[0].Instances[0].NetworkInterfaces[0].Attachment.DeviceIndex = &nonzeroDeviceIndex
+				mock.EXPECT().DescribeInstances(&ec2.DescribeInstancesInput{
+					InstanceIds: []*string{&testInstance},
+				}).Return(&output, nil)
+			},
+			expectErr: "verifying the EC2 instance's NetworkInterface[0].DeviceIndex is 0",
+		},
+		{
+			desc: "success, default template",
+			mockExpect: func(mock *mock_aws.MockEC2Client) {
+				output := getDefaultDescribeInstancesOutput()
+				output.Reservations[0].Instances[0].RootDeviceType = &instanceStoreType
+				output.Reservations[0].Instances[0].NetworkInterfaces[0].Attachment.DeviceIndex = &zeroDeviceIndex
+				mock.EXPECT().DescribeInstances(&ec2.DescribeInstancesInput{
+					InstanceIds: []*string{&testInstance},
+				}).Return(&output, nil)
+			},
+			expectID: "spiffe://example.org/spire/agent/aws_iid/test-account/test-region/test-instance",
+		},
+		{
+			desc: "success, different template",
+			mockExpect: func(mock *mock_aws.MockEC2Client) {
+				output := getDefaultDescribeInstancesOutput()
+				output.Reservations[0].Instances[0].RootDeviceType = &instanceStoreType
+				output.Reservations[0].Instances[0].NetworkInterfaces[0].Attachment.DeviceIndex = &zeroDeviceIndex
+				mock.EXPECT().DescribeInstances(&ec2.DescribeInstancesInput{
+					InstanceIds: []*string{&testInstance},
+				}).Return(&output, nil)
+			},
+			replacementTemplate: template.Must(template.New("agent-svid").Parse("{{ .PluginName}}/{{ .Region }}/{{ .AccountID }}/{{ .InstanceID }}")),
+			expectID:            "spiffe://example.org/spire/agent/aws_iid/test-region/test-account/test-instance",
+		},
+	}
+
+	for _, tt := range tests {
+		s.T().Run(tt.desc, func(t *testing.T) {
+			mockCtl := gomock.NewController(s.T())
+			defer mockCtl.Finish()
+
+			ec2Client := mock_aws.NewMockEC2Client(mockCtl)
+
+			originalGetEC2Client := s.plugin.getClient
+			defer func() {
+				s.plugin.getClient = originalGetEC2Client
+			}()
+			mockGetEC2Client := func(p client.ConfigProvider, cfgs ...*awssdk.Config) EC2Client {
+				return ec2Client
+			}
+			s.plugin.getClient = mockGetEC2Client
+			tt.mockExpect(ec2Client)
+
+			if tt.replacementTemplate != nil {
+				originalTemplate := s.plugin.config.pathTemplate
+				defer func() {
+					s.plugin.config.pathTemplate = originalTemplate
+				}()
+				s.plugin.config.pathTemplate = tt.replacementTemplate
+			}
+
+			data := &common.AttestationData{
+				Type: aws.PluginName,
+				Data: s.iidAttestationDataToBytes(*s.buildDefaultIIDAttestationData()),
+			}
+
+			resp, err := s.attest(&nodeattestor.AttestRequest{
+				AttestationData: data,
+			})
+
+			if tt.expectErr != "" {
+				s.Nil(resp)
```

Comment:
using the Suite methods doesn't work well within a subtest (e.g. testing.T.Run). you should use the `assert` or `require` package functions directly with the `*testing.T` provided by `Run()`.